### ISSUE TITLE
Would you accept an `http.Handler` implementation?

### ISSUE THREAD

**Issue by codyaray** [getkin/kin-openapi#195]
This is really what I was looking for in openapi-validation. This makes for an extremely easy integration path with existing applications. This is already how some other services like 	OpenCensus integrate as well (see `ochttp` in the sample `main` below).

This would invoke the request and response validators automatically. It would require a `ValidationErrorEncoder` and I think we can provide a [good default implementation](https://github.com/getkin/kin-openapi/issues/194) for that as well.

Here's an example of how an integration may look:
```
h, err := app.NewHTTPHandler(...)
check("unable to create handler", err)

// wrap with openapi validation handler
h = &openapi3filter.ValidationHandler{
  Handler: h,
  SwaggerFile: "docs/openapi.yaml",
  ErrorEncoder: openapi3filter.ValidationsErrorEncoder{Encoder: myAppEncoder}
}

// wrap by ochttp handler
h = &ochttp.Handler{Handler: h}

// start server
s := &gohttp.Server{Addr: *httpAddr, Handler: h}
ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
defer s.Shutdown(ctx)
defer cancel()

level.Info(l).Log("msg", fmt.Sprintf("Starting http server at addr: %s", *httpAddr))
errc <- s.ListenAndServe()
```

**Comment by fenollp** [comment]
An `http.Handler` implementation makes total sense.
I'd also accept an implementation of https://golang.org/pkg/net/http/#RoundTripper

**Comment by codyaray** [comment]
Addressed in #197
